modules/collection/programs/fish.nix

```
@@ -7,31 +7,27 @@
   inherit (lib.options) literalExpression mkOption mkEnableOption mkPackageOption;
   inherit (lib.strings) typeOf concatMapAttrsStringSep concatMapStringsSep splitString escapeShellArg;
   inherit (lib.modules) mkIf;
-  inherit (lib.types) either str path oneOf attrsOf nullOr;
+  inherit (lib.types) lines path oneOf attrsOf;
   inherit (lib.attrsets) mapAttrs' nameValuePair isDerivation filterAttrs attrValues;
   inherit (lib.lists) any filter optionals;
   inherit (lib.trivial) pathExists;
 
   cfg = config.rum.programs.fish;
   env = config.environment.sessionVariables;
 
-  toFish = strOrPath: fileName:
-    if (typeOf strOrPath) == "string"
-    then pkgs.writers.writeFish fileName strOrPath
-    else if isDerivation strOrPath
-    then strOrPath
-    else throw "Input is of invalid type ${typeOf strOrPath}, expected `path` or `string`.";
-
-  toFishFunc = strOrPath: funcName:
-    toFish (
-      if (typeOf strOrPath) == "string"
-      then ''
+  toFish = body: fileName: pkgs.writers.writeFish fileName body;
```

This is unnecessary, just use an inherit and use writeFish directly.
```suggestion
inherit (pkgs.writers) writeFish;
```

---

Done.